tests/State/HubTest.php

```diff
@@ -167,6 +167,35 @@ public function testWithScope(): void
         $this->assertTrue($callbackInvoked);
     }
 
+    public function testWithScopeWithReturnValue(): void
```

Comment:
This test can be merged with the previous one, as suggested for the other similar places